tools/tpm2_pcrreset.c

```diff
@@ -10,71 +10,176 @@
 
 typedef struct tpm_pcr_reset_ctx tpm_pcr_reset_ctx;
 struct tpm_pcr_reset_ctx {
+    /*
+     * Inputs
+     */
     bool pcr_list[TPM2_MAX_PCRS];
+
+    /*
+     * Outputs
+     */
 };
 
 static tpm_pcr_reset_ctx ctx;
 
-static tool_rc pcr_reset_one(ESYS_CONTEXT *ectx, TPMI_DH_PCR pcr_index) {
-
-    tool_rc rc = tpm2_pcr_reset(ectx, pcr_index);
-    if (rc != tool_rc_success) {
-        LOG_ERR("Could not reset PCR index: %d", pcr_index);
-    }
-
-    return rc;
-}
-
 static tool_rc pcr_reset(ESYS_CONTEXT *ectx) {
-    size_t i;
 
+    size_t i;
     for (i = 0; i < TPM2_MAX_PCRS; i++) {
-        if (!ctx.pcr_list[i])
+        if (!ctx.pcr_list[i]) {
             continue;
+        }
 
-        tool_rc rc = pcr_reset_one(ectx, i);
+        tool_rc rc = tpm2_pcr_reset(ectx, i);
         if (rc != tool_rc_success) {
+            LOG_ERR("Could not reset PCR index: %ld", i);
             return rc;
         }
     }
 
     return tool_rc_success;
 }
 
-static bool on_arg(int argc, char** argv) {
-    int i;
-    uint32_t pcr;
+static tool_rc process_outputs(ESYS_CONTEXT *ectx) {
 
-    memset(ctx.pcr_list, 0, TPM2_MAX_PCRS);
+    UNUSED(ectx);
+
+    /*
+     * 1. Outputs that do not require TPM2_CC_<command> dispatch
+     */
+
+    /*
+     * 2. Outputs generated after TPM2_CC_<command> dispatch
+     */
+
+    return tool_rc_success;
+}
```

Comment:
So here and in a few other places we have functions that get defined but just return success without doing anything, is it worth keeping these for future expansion in connor's cpHash work or is it better to not clutter the code and delete them?

Reply:
Then there is rpHash, session support, auditing, tcti=none for cpHash. With the template it makes it a little easier to replicate the patches. In short term, it may seem a clutter for sure.